modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/ClientFailoverSocket.cs

```diff
@@ -508,7 +508,12 @@ private void OnNewDefaultConnection()
                 );
             }
 
-            if (!_socket.Features.HasFeature(ClientBitmaskFeature.ClusterGroupGetNodesEndpoints))
+            if (!_config.EnablePartitionAwareness || !_config.EnableClusterDiscovery)
```

Comment:
What if user wants Cluster Discovery but without Partition Awareness? Just for failover?

Reply:
Good point. I've discarded this use case as weird initially, but there in no reason not to support it.

Fixed.